uaclient/security_status.py

```diff
@@ -129,23 +129,35 @@ def get_update_status(service_name: str, ua_info: Dict[str, Any]) -> str:
 def filter_security_updates(
     packages: List[apt_package.Package],
 ) -> DefaultDict[str, List[Tuple[apt_package.Version, str]]]:
-    """Filters a list of packages looking for available security updates.
+    """Filters a list of packages looking for available updates.
 
-    Checks if the package has a greater version available, and if the origin of
-    this version matches any of the series' security repositories.
+    All versions greater than the installed one are reported, based on where
+    it is provided, including ESM pockets, excluding backports.
     """
     result = defaultdict(list)
 
     for package in packages:
-        for version in package.versions:
-            if version > package.installed:
-                for origin in version.origins:
-                    service = get_origin_information_to_service_map().get(
-                        (origin.origin, origin.archive)
-                    )
-                    if service:
-                        result[service].append((version, origin.site))
-                        break  # No need to loop through all the origins
+        if package.is_installed:
+            for version in package.versions:
+                if version > package.installed:
+                    is_security = False
+                    for origin in version.origins:
+                        service = get_origin_information_to_service_map().get(
+                            (origin.origin, origin.archive)
+                        )
+                        if service:
+                            result[service].append((version, origin.site))
+                            is_security = True
+                            break  # No need to loop through all the origins
+                    # Also no need to report backports at least for now...
+                    expected_origin = version.origins[0]
```

Comment:
Why are using the first origin here ?

Reply:
Because at this point, we know it is a regular update, which will be either on main/universe/multiverse/restricted but not in more than one of them. So, the first origin is where APT is going to download it from, and given it is not security, we're fine with it.